servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -657,7 +657,12 @@ AABB MeshStorage::mesh_get_aabb(RID p_mesh, RID p_skeleton) {
 
 	Skeleton *skeleton = skeleton_owner.get_or_null(p_skeleton);
 
-	if (!skeleton || skeleton->size == 0 || mesh->skeleton_aabb_version == skeleton->version) {
+	if (!skeleton || skeleton->size == 0) {
+		return mesh->aabb;
+	}
+
+	// A mesh can be shared by multiple skeletons and we need to avoid using the AABB from a different skeleton
+	if (mesh->skeleton_aabb_version == skeleton->version && mesh->skeleton_aabb_rid == p_skeleton) {
```

Comment:
```suggestion
	// A mesh can be shared by multiple skeletons and we need to avoid using the AABB from a different skeleton.
	if (!skeleton || skeleton->size == 0 || (mesh->skeleton_aabb_version == skeleton->version && mesh->skeleton_aabb_rid == p_skeleton)) {
```
I'd say this doesn't require to different if statements doing the same thing

Reply:
I feel two separate if statements is easier to read: it logically separates between meshes that aren't connected to a skeleton and the re-use of a previously computed skeleton AABB. This is more of a style thing though and if combining them in a single statement fits better with existing code I can change it

Reply:
I don't see it conveying anything specific, it returns the same value and separating it into different blocks makes me confused as to why the second block is separate, when it returns the same value, and makes me confused as to if for example the value is modified

Reply:
Okay, I'll combine them as you show above and add the missing period at the end of the comment